src/lib/airbrake.js

```diff
@@ -0,0 +1,142 @@
+const { formatWithOptions, inspect } = require('util')
+const { Notifier } = require('@airbrake/node')
+
+const INSPECT_OPTS = {
+  depth: null,
+  maxStringLength: null,
+  maxArrayLength: null,
+  breakLength: null,
+  compact: true,
+  showHidden: true,
+  errorNotifications: true
+}
+
+class AirbrakeClient {
+  /**
+   * Initializes the Airbrake client and intercepts `console.warn` and `console.error`
+   * to send notifications to Airbrake/Errbit.
+   *
+   * If the required environment variables are not set, Airbrake will not be initialised.
+   */
+  constructor () {
+    /** @type {Notifier} */
+    this.airbrake = null
+
+    const { AIRBRAKE_PROJECT_KEY, AIRBRAKE_HOST } = process.env
+
+    if (!AIRBRAKE_PROJECT_KEY || !AIRBRAKE_HOST) {
+      console.info('[Airbrake] Not initialised. Missing environment variables:', {
+        AIRBRAKE_PROJECT_KEY: !!AIRBRAKE_PROJECT_KEY,
+        AIRBRAKE_HOST: !!AIRBRAKE_HOST
+      })
+      return
+    }
+
+    this.airbrake = new Notifier({
+      projectId: 4,
+      projectKey: process.env.AIRBRAKE_PROJECT_KEY,
+      host: process.env.AIRBRAKE_HOST,
+      environment: process.env.NODE_ENV,
+      errorNotifications: true,
+      performanceStats: false,
+      remoteConfig: false
+    })
+
+    const originalError = console.error.bind(console)
+    const originalWarn = console.warn.bind(console)
+
+    console.error = (...args) => {
+      this.reportToAirbrake('error', ...args)
+      originalError(...args)
+    }
+
+    console.warn = (...args) => {
+      this.reportToAirbrake('warn', ...args)
+      originalWarn(...args)
+    }
+
+    // Ensure uncaught errors are logged
+    process.on('uncaughtExceptionMonitor', originalError)
+
+    // Override Airbrake's uncaughtException/unhandledRejection handlers to flush before exit
+    const flushAndExit = async () => {
+      await this.airbrake.flush()
+      process.exit(1)
+    }
+
+    process.on('uncaughtException', flushAndExit)
+    process.on('unhandledRejection', flushAndExit)
+  }
+
+  /**
+   * Reports a console method invocation to Airbrake.
+   *
+   * @param {string} method - The console method being called (e.g., "warn" or "error").
+   * @param {...any} args - The arguments passed to the console method.
+   * @returns {void}
+   */
+  reportToAirbrake (method, ...args) {
+    const error =
+    args.find((arg) => arg instanceof Error) ??
+    new Error(formatWithOptions(INSPECT_OPTS, ...args))
+
+    const request = args.find(
+      (arg) => arg && typeof arg === 'object' && 'headers' in arg
+    )
+
+    // notify returns a promise, but we do not await it
+    this.airbrake.notify({
```

Comment:
Doesn't this need to check that `this.airbrake` has been defined, like you do in `attachAirbrakeToDebugLogger` and `flush`?